src/test/java/org/jabref/gui/externalfiles/FileFilterUtilsTest.java

```diff
@@ -0,0 +1,207 @@
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+public class FileFilterUtilsTest {
+
+    private final FileFilterUtils fileFilterUtils = new FileFilterUtils();
+    private final LocalDateTime time = LocalDateTime.now();
+
+    @Test
+    public void isDuringLastDayNegativeTest() {
+        assertEquals(fileFilterUtils.isDuringLastDay(time.minusHours(24)), false);
+    }
+
+    @Test
+    public void isDuringLastDayPositiveTest() {
+        assertEquals(fileFilterUtils.isDuringLastDay(time.minusHours(23)), true);
+    }
+
+    @Test
+    public void isDuringLastWeekNegativeTest() {
+        assertEquals(fileFilterUtils.isDuringLastWeek(time.minusDays(7)), false);
+    }
+
+    @Test
+    public void isDuringLastWeekPositiveTest() {
+        assertEquals(fileFilterUtils.isDuringLastWeek(time.minusDays(6).minusHours(23)), true);
+    }
+
+    @Test
+    public void isDuringLastMonthNegativeTest() {
+        assertEquals(fileFilterUtils.isDuringLastMonth(time.minusDays(30)), false);
+    }
+
+    @Test
+    public void isDuringLastMonthPositiveTest() {
+        assertEquals(fileFilterUtils.isDuringLastMonth(time.minusDays(29).minusHours(23)), true);
+    }
+
+    @Test
+    public void isDuringLastYearNegativeTest() {
+        assertEquals(fileFilterUtils.isDuringLastYear(time.minusDays(365)), false);
+    }
+
+    @Test
+    public void isDuringLastYearPositiveTest() {
+        assertEquals(fileFilterUtils.isDuringLastYear(time.minusDays(364).minusHours(23)), true);
+    }
+
+    @Nested
+    class SortingTests {
+
+        private List<Path> files = new ArrayList<Path>();
+        private List<String> expectedSortByDateAscending = new ArrayList<String>();
+        private List<String> expectedSortByDateDescending = new ArrayList<String>();
+        private List<String> wrongOrder = new ArrayList<String>();
+
+        /* Initialize the directory and files used in the sorting tests, and change their last edited dates. */
+        @BeforeEach
+        public void setUp() throws Exception {
+            String dirPath = "src/test/resources/unlinkedFiles";
+
+            Path path = Path.of(dirPath);
+            Files.createDirectory(path);
+    
+            Path firstPath = Path.of(dirPath.concat("/firstFile.pdf"));
+            Path secondPath = Path.of(dirPath.concat("/secondFile.pdf"));
+            Path thirdPath = Path.of(dirPath.concat("/thirdFile.pdf"));
+            Path fourthPath = Path.of(dirPath.concat("/fourthFile.pdf"));
+    
+            Files.createFile(firstPath);
+            Files.createFile(secondPath);
+            Files.createFile(thirdPath);
+            Files.createFile(fourthPath);
+
+            // change the files last edited times.
+            File firstFile = new File(firstPath.toString());
```

Comment:
use the nio method so you don't the 
 https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#createFile(java.nio.file.Path,%20java.nio.file.attribute.FileAttribute...)

Reply:
Hello @Siedlerchr we tried to resolve this issue on this commit: a495c1647800be62cc7d4c18f9f2d94097d05950